### PR TITLE
[CM-1265] Improve events to expose message

### PR DESCRIPTION
**Pull request by Sathyan-KM**
## Summary
- Exposed message through events
- In current version we have single callback for all events now  created callback for each events. It will remove the parsing the dictionary on every events for customers.

### CODE REVIEW

File: Sources/Controllers/ALKConversationViewController.swift [hunk 1]
```diff
@@ -1317,6 +1317,8 @@ open class ALKConversationViewController: ALKBaseViewController, Localizable {
         default:
             print("Do nothing")
         }
+        guard let conversationId = message.channelKey else {return}
+        ALKCustomEventHandler.shared.publish(triggeredEvent: CustomEvent.richMessageClick, data:  ["conversationId": conversationId,"action": action, "type": type])
```
Comment (mukesh-komm) [inline Sources/Controllers/ALKConversationViewController.swift:1321]: How is this different from the above click event?

File: Sources/CustomEvent/CustomEvent.swift [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
-    case rateConversationEmotionsClick = "ON_RATE_CONVERSATION_EMOTIONS_CLICK"
     case submitRatingClick = "ON_SUBMIT_RATING_CLICK"
     case restartConversationClick = "ON_RESTART_CONVERSATION_CLICK"
     case richMessageClick = "ON_RICH_MESSAGE_CLICK"
-    case resolveConversation = "ON_CONVERSATION_RESOLVE"
     case conversationBackPress = "ON_CONVERSATION_BACK_CLICK"
     case conversationListBackPress = "ON_CONVERSATION_LIST_BACK_CLICK"
     case messageReceive = "ON_MESSAGE_RECEIVE"
+    // TODO: ADD THESE EVENTS(attachmentClick,voiceClick, locationClick, rateConversationEmotionsClick) WITH PROPER DATA
+    // TODO: FIX FLOW: resolveConversation
```
Comment (mukesh-komm) [inline Sources/CustomEvent/CustomEvent.swift:19]: We can file issues for this and remove from here if they are not needed.

File: Sources/CustomEvent/CustomEvent.swift [hunk 3]
```diff
@@ -1,25 +1,20 @@
...
+// Created by Sathyan Elangovan on 09/12/21.
 
 import Foundation
 
 public enum CustomEvent: String {
     case messageSend = "ON_MESSAGE_SEND"
     case faqClick = "ON_FAQ_CLICK"
-    case notificationClick = "ON_NOTIFICATION_CLICK"
     case newConversation = "ON_START_NEW_CONVERSATION_CLICK"
-    case attachmentClick = "ON_ATTACHMENT_ICON_CLICK"
```
Comment (mukesh-komm) [inline Sources/CustomEvent/CustomEvent.swift:13]: What if someone is using these?

File: Sources/CustomEvent/CustomEvent.swift (hunk 2 above)
Comment (Sathyan-KM) [reply Sources/CustomEvent/CustomEvent.swift]: removed & filed issue

File: Sources/Controllers/ALKConversationViewController.swift (hunk 1 above)
Comment (Sathyan-KM) [reply Sources/Controllers/ALKConversationViewController.swift:1321]: above one was supposed to be removed.removed it

File: Sources/CustomEvent/CustomEvent.swift (hunk 3 above)
Comment (Sathyan-KM) [reply Sources/CustomEvent/CustomEvent.swift:13]: once they upgrade to new version, these events will not considered.Wr have removed old callback & new callbacks has to be implemented .Will add the migration doc in release notes as well as in our docs

Comment (Sathyan-KM) [reply Sources/CustomEvent/CustomEvent.swift:13]: we have another release in the next week, will add those events there

**Review by mukesh-komm — APPROVED** [review]

**Pull request merged**